code/scripting/api/objs/shipwepselect.cpp

```diff
@@ -218,19 +279,37 @@ ADE_OBJ(l_Loadout_Ship, int, "loadout_ship", "Loadout handle");
 
 ADE_VIRTVAR(ShipClassIndex,
 	l_Loadout_Ship,
-	nullptr,
-	"The index of the Ship Class",
+	"number",
+	"The index of the Ship Class. When setting the ship class this will also set the weapons to empty slots. Use "
+	".Weapons and .Amounts to set those afterwards. Set to -1 to empty the slot.",
 	"number",
 	"The index or nil if handle is invalid")
 {
 	int current;
-	if (!ade_get_args(L, "o", l_Loadout_Ship.Get(&current))) {
+	int shipIndex;
+	if (!ade_get_args(L, "o|i", l_Loadout_Ship.Get(&current), &shipIndex)) {
 		return ADE_RETURN_NIL;
 	}
 
-	// Will eventually be writable so that Lua can manipulate the player loadout!
+	shipIndex--; //Convert from Lua
+
 	if (ADE_SETTING_VAR) {
-		LuaError(L, "This property is read only.");
+		if (shipIndex > -1) {
+			// If we're not trying to empty the slot then make sure it's a valid ship index
+			if (shipIndex > ship_info_size()) {
+				LuaError(L, "The provided ship index is invalid!");
+			} else {
+				Wss_slots[current].ship_class = shipIndex;
+				//Reset all currently loaded weapons here
+				int i;
+				for (i = 0; i < MAX_SHIP_WEAPONS; i++) {
```

Comment:
Just combine these and add the ``int`` in front of the ``i = 0`` in the for loop instead of using an extra line to allocate the int

Reply:
Done